Add TFC example (#23)
infracost/actions: examples/terraform-cloud/README.md

> # Terraform cloud

Reviewer: ```suggestion
# Terraform Cloud/Enterprise

This example shows how to run Infracost actions with Terraform Cloud and Terraform Enterprise. It assumes you have set a GitHub repo secret for the Terraform Cloud token (`TFC_TOKEN`). This token is used by the Infracost CLI to fetch the plan JSON from Terraform Cloud and use it to generate the cost estimate comment.

In the future, we'll add an example of how you can trigger the Infracost actions from Terraform Cloud's GitHub status checks.
```

Can you please edit the description to clarify which plan is fetched to generate the cost estimate? Or is a new speculative plan generated for each pull request?

[//]: <> (BEGIN EXAMPLE)
```yml
name: Terraform cloud
on: [pull_request]

jobs:
  terraform-cloud:
    name: Terraform cloud
    runs-on: ubuntu-latest

    steps:
      - uses: actions/checkout@v2
      
      - name: Install terraform
        uses: hashicorp/setup-terraform@v1
        with:
          terraform_wrapper: false # This is required so the `terraform show` command outputs valid JSON

>           cli_config_credentials_token: $${{ secrets.TFC_TOKEN }}

Reviewer: ```suggestion
          cli_config_credentials_token: $${{ secrets.TFC_TOKEN }}
          # cli_config_credentials_hostname: https://my_tfe_host.com # For Terraform Enterprise users only
```

      - name: Setup Infracost
        uses: infracost/actions/setup@v1
        with:
          api-key: ${{ secrets.INFRACOST_API_KEY }}
          
      - name: Run Infracost
        run: infracost breakdown --path=examples/terraform-cloud/code --format=json --out-file=/tmp/infracost.json
        env:

>           INFRACOST_TERRAFORM_CLOUD_TOKEN: ${{ secrets.TFC_TOKEN }} # TODO: can be removed once https://github.com/infracost/infracost/pull/1148 is released

Reviewer: ```suggestion
          # TODO: the following two envs be removed once https://github.com/infracost/infracost/pull/1148 is released in v0.9.14 of the CLI (https://github.com/infracost/infracost/releases)
          INFRACOST_TERRAFORM_CLOUD_TOKEN: ${{ secrets.TFC_TOKEN }} 
          # INFRACOST_TERRAFORM_CLOUD_HOST: https://my_tfe_host.com # For Terraform Enterprise users only.
```
